torchtune/training/memory.py

```diff
@@ -288,19 +299,19 @@ def log_memory_stats(
 ) -> None:
     """
     Logs a dict containing memory stats to the logger. ``stats`` should contain the fields
-    ``peak_memory_active``, ``peak_memory_alloc``, and ``peak_memory_reserved`` as
+    ``peak_memory_active``, ``peak_memory_alloc``, and ``peak_memory_reserved`` (optional) as
     returned by :func:`torchtune.training.get_memory_stats`.
 
     Args:
         stats (Dict[str, float]): A dictionary containing the peak memory active, peak memory
-            allocated, and peak memory reserved stats.
+            allocated, and peak memory reserved (optional) stats.
         message (str): An optional message to prepend to the log output.
             Defaults to "Memory stats after model init:"
     """
     device_support = get_device_support()
     _log.info(
```

Comment:
Let's loop over the dictionary items so we log the key and the value. This way, we can omit the peak_memory_reserved if it doesn't exist at all.